app/ui/src/app/customizations/customizations.module.ts

```diff
@@ -25,7 +25,8 @@ const routes: Routes = [
       },
       {
         path: 'api-connector',
-        component: ApiConnectorListComponent
+        component: ApiConnectorListComponent,
+        canActivate: [ApiConnectorLazyLoaderGuard]
```

Comment:
Wow we even have our own guard now!

Reply:
Yep, that guard takes care of checking if the ApiConenctors have been fetched from the API already and will trigger the fetch action if not, which will set the overall state in "loading", so when the user makes it to ANY page requiring that data, the loading state will be already on and the connectors will be rendered once loaded. Next time you return to that same page, no fetch (hence no http call) action will be triggered.

Try it out yourself: Go to the API Connectors page with the Networks tab open. You'll see a http request being made. Go back, then return to the page: No more http requests are triggered, since the information is now fetched from the store.

Try this out now: Go straight to the any connector detail view. Hard reload the page and see how you're fetching ALL the connectors, including the one you need. Now click on the lsit link and see that you're no longer requesting anything form the backend.

Last test: Go to the wizard and create a new connector (use this URL for fetching a awagger file: `http://petstore.swagger.io/v2/swagger.json`). Complete the steps and see how the connectors are re-fetched once the create process is done. Next time you access the list page, you're newly created connector will be already there.

Bottom line is that, by doing this, we:

* Reduce the backend overhead by minimizing HTTP calls
* Pages speed becomes blazing fast since all the information is reactively returned from the store.

Probs some people will thing the above is trivial, but think for a minute this site had millions of uniques a day hammering the API. This kinda architecture ensures site resilience and stability.

Reply:
Also, it's worth mentioned the change applied in the overall component controller implementation. Relying on a Store reduces the boilerplate to the max, since components become mere consumers of the store and action dispatchers, not requiring any further logic in their implementation.